typescript/api/services/TranslationService.ts

```diff
@@ -34,49 +36,279 @@ export module Services {
+        resources[lng] = {};
+        for (const ns of namespaces) {
+          try {
+            // Prefer explicit fields query
+            let bundle = await I18nBundle.findOne({ branding: brandingId, locale: lng, namespace: ns });
+            if (!bundle) {
+              // Fallback to uid-based lookup if needed
+              const uid = `${brandingId}:${lng}:${ns}`;
+              bundle = await I18nBundle.findOne({ uid });
+            }
+            const data = (bundle?.data && typeof bundle.data === 'object') ? { ...bundle.data } : {};
+            if (data && typeof data === 'object' && data._meta) {
+              // strip metadata from runtime resources
+              delete (data as any)._meta;
+            }
+            resources[lng][ns] = data || {};
+          } catch (e) {
+            // Note: Keeping this as verbose since it's already a low-level diagnostic message
+            sails.log.verbose('[TranslationService] Failed to load bundle', brandingId, lng, ns, (e as Error)?.message || e);
+            resources[lng][ns] = {};
+          }
+        }
+      }
+      return resources;
+    }
+
+    public t(key, context = undefined, langCode: string = 'en', brandingName: string = 'default') {
+      const brand = BrandingService.getBrand(brandingName);
+
+      const i18nextInstance = this.i18nextInstances[brand.id];
+
+      if (!i18nextInstance) {
+        this.logger.warn(`No i18next instance found for brand name: ${brandingName}, branding id: ${brand.id}, falling back to key`);
+        return key;
+      }
+      return i18nextInstance.getFixedT(langCode)(key, context);
     }
 
     public tInter(key, context = null, langCode:string = 'en') {
       return this.t(key, context, langCode);
     }
 
-    public reloadResources() {
-      //@ts-ignore
-      i18next.reloadResources();
+    public async reloadResources(brandingId?: string) {
+      // Re-create i18next instances so deletions / prunes are reflected (addResourceBundle can't remove keys)
+      const brandingsToReload = brandingId ? [brandingId] : Object.keys(this.i18nextInstances);
+      for (const bId of brandingsToReload) {
+        try {
+          await this.clearInstances(bId);
+          const branding = BrandingService.getBrand(bId);
+          if (!branding) continue;
+          await this.getI18nextForBranding(branding);
+          this.logger.debug(`Re-created i18next instance for branding: ${bId}`);
+        } catch (e) {
+          this.logger.warn(`Failed to re-create i18next for branding ${bId}:`, (e as Error)?.message || e);
+        }
+      }
     }
 
-    public handle(req, res, next) {
+    /**
+     * Get available languages for a specific branding from DB
+     */
+    public async getAvailableLanguagesForBranding(branding: any): Promise<string[]> {
+      try {
+        if (!branding) {
+          this.logger.warn('No branding provided, using config fallback');
+          return sails.config.i18n.next.init.supportedLngs || ['en'];
```

Comment:
This (`sails.config.i18n.next.init.supportedLngs`) is accessed later using optional chaining - could this access be null as well? Should it use optional chaining too?